From 64a16b95a9b188a280df81c220154ef379e15eb2 Mon Sep 17 00:00:00 2001
From: Mitch Talmadge <mitcht@aptitekk.com>
Date: Wed, 30 Jul 2014 03:14:05 -0600
Subject: [PATCH 071/101] Fixed a bug with TeamScreen putting the
 AutoUpdater.jar inside /updates/. Also made TeamScreen delete the
 AutoUpdater.jar and /updates/ folder upon startup.

---
 .../TeamScreen/AutoUpdater/AutoUpdater.java   | 29 ++++++++++++-------
 .../AutoUpdaterDownloadThread.java            |  9 +++++-
 .../aptitekk/TeamScreen/TeamScreenDaemon.java |  5 ++--
 3 files changed, 29 insertions(+), 14 deletions(-)

diff --git a/TeamScreen/src/com/aptitekk/TeamScreen/AutoUpdater/AutoUpdater.java b/TeamScreen/src/com/aptitekk/TeamScreen/AutoUpdater/AutoUpdater.java
index 88ec01d..fe0ab9a 100644
--- a/TeamScreen/src/com/aptitekk/TeamScreen/AutoUpdater/AutoUpdater.java
+++ b/TeamScreen/src/com/aptitekk/TeamScreen/AutoUpdater/AutoUpdater.java
@@ -1,7 +1,5 @@
 package com.aptitekk.TeamScreen.AutoUpdater;
 
-import java.awt.Color;
-import java.awt.Font;
 import java.awt.Image;
 import java.io.File;
 import java.io.FileInputStream;
@@ -15,11 +13,6 @@ import java.net.URL;
 import java.net.URLConnection;
 import java.util.ArrayList;
 
-import javax.net.ssl.HttpsURLConnection;
-import javax.swing.SwingUtilities;
-import javax.swing.UIManager;
-import javax.swing.border.EmptyBorder;
-
 import org.apache.commons.io.FileUtils;
 
 public class AutoUpdater
@@ -36,11 +29,20 @@ public class AutoUpdater
     private AutoUpdaterDownloadThread autoUpdaterDownloadThread;
     private static final Object synchronizedObject = new Object();
     private Handshake handShake;
+    private boolean askUserToUpdate;
     public static Image windowIcon;
     private static ArrayList<UpdaterListener> updateListeners = new ArrayList<UpdaterListener>();
         
-    public AutoUpdater()
+    /**
+     * Creates a new AutoUpdater. 
+     * @param askUserToUpdate - Display a ChangeLog (true), or skip ChangeLog and download immediately (false)
+     * @param windowIcon - The icon file to be displayed in Task Bar.
+     */
+    public AutoUpdater(boolean askUserToUpdate, Image windowIcon)
     {
+	deleteUpdatesFolder();
+	this.askUserToUpdate = askUserToUpdate;
+	AutoUpdater.windowIcon = windowIcon;
 	System.setProperty("javax.net.ssl.trustStore",getAbsDir()+"/secure/cacerts");
 	System.setProperty("javax.net.ssl.trustStorePassword","CpAhLLp94j");
 	this.loadProperties();
@@ -102,7 +104,7 @@ public class AutoUpdater
 	    String changeLog = this.sendPOSTGetString(
 		    AutoUpdater.updaterURLPrefix
 		    + AutoUpdater.updaterChangeLogSuffix, POSTData);
-	    if(this.askUserToUpdate(changeLog, amountToUpdate))
+	    if(!askUserToUpdate || this.askUserToUpdate(changeLog, amountToUpdate))
 	    {
 		System.out.println("User said Yes. Generating Handshake...");
 		if(generateHandshake())
@@ -226,6 +228,9 @@ public class AutoUpdater
 		e.printStackTrace();
 	    }
 	}
+	File autoUpdater = new File(getAbsDir() + "/AutoUpdater.jar");
+	if(autoUpdater.exists())
+	    autoUpdater.delete();
     }
     
     public String getCurrentVersionName()
@@ -261,7 +266,11 @@ public class AutoUpdater
 	}
     }
     
-    public void addUpdateListener(UpdaterListener listener)
+    /**
+     * Adds an UpdaterListener (For shutting down, etc)
+     * @param listener - The UpdaterListener.
+     */
+    public void addUpdaterListener(UpdaterListener listener)
     {
 	if(listener != null)
 	{
diff --git a/TeamScreen/src/com/aptitekk/TeamScreen/AutoUpdater/AutoUpdaterDownloadThread.java b/TeamScreen/src/com/aptitekk/TeamScreen/AutoUpdater/AutoUpdaterDownloadThread.java
index 03c7b3a..3160aa2 100644
--- a/TeamScreen/src/com/aptitekk/TeamScreen/AutoUpdater/AutoUpdaterDownloadThread.java
+++ b/TeamScreen/src/com/aptitekk/TeamScreen/AutoUpdater/AutoUpdaterDownloadThread.java
@@ -26,6 +26,13 @@ public class AutoUpdaterDownloadThread extends Thread
     public void run()
     {
 	URLUtils.addProgressListener(new DownloadDialog(amountToUpdate+1));
+	try
+	{
+	    Thread.sleep(500);
+	}
+	catch(InterruptedException e)
+	{
+	}
 	if(downloadUpdate(-1))
 	{
 	    for(int i = 1; i <= amountToUpdate; i++)
@@ -63,7 +70,7 @@ public class AutoUpdaterDownloadThread extends Thread
 	    String filename = contentDisposition.substring(contentDisposition.indexOf("\"")+1, contentDisposition.lastIndexOf("\""));
 	    
 	    File update = URLUtils.getFileFromConnection(connection,
-		    AutoUpdater.getAbsDir() + "/updates/" + filename);
+		    AutoUpdater.getAbsDir() + (versionID == -1 ? "/" : "/updates/") + filename);
 	    if(update == null)
 		return false;
 	    return true;
diff --git a/TeamScreen/src/com/aptitekk/TeamScreen/TeamScreenDaemon.java b/TeamScreen/src/com/aptitekk/TeamScreen/TeamScreenDaemon.java
index 95c4d53..9f5c14d 100644
--- a/TeamScreen/src/com/aptitekk/TeamScreen/TeamScreenDaemon.java
+++ b/TeamScreen/src/com/aptitekk/TeamScreen/TeamScreenDaemon.java
@@ -142,9 +142,8 @@ public class TeamScreenDaemon implements KeyListener, HotkeyListener, UpdaterLis
 	    
 	    displayPopup(trayIcon, "Checking For Updates...", "TeamScreen is now checking for updates...", TrayIcon.MessageType.INFO);
 	    
-	    AutoUpdater.windowIcon = FolderManager.createImage("/images/TeamScreen-64px.png", "tray icon");
-	    AutoUpdater updater = new AutoUpdater();
-	    updater.addUpdateListener(this);
+	    AutoUpdater updater = new AutoUpdater(true, FolderManager.createImage("/images/TeamScreen-64px.png", "tray icon"));
+	    updater.addUpdaterListener(this);
 	    updater.checkForUpdates();
 	    
 	    this.netHandler = new NetHandler(this);
-- 
2.39.5 (Apple Git-154)

